### DISCUSSION TITLE
[NEW FEATURE]: Summary of important permissions

### DISCUSSION THREAD

**Discussion by TheRealMystic** [d4rken-org/permission-pilot#258] [Ideas]
When clicking on an app in the Apps tab, the new page shows all the permissions for the app.

Is it possible to provide a summary of those 30 odd permissions in a very simplified way, something like this:

<img width="359" alt="Summary" src="https://user-images.githubusercontent.com/54448474/179664513-546e646e-707f-4d7b-97bb-01e59cca7e77.png">

This will be followed by all the permissions in detail, as it exists now.

The idea is to give the user a quick glance of the status of all permissions that is available for modification within System Settings.

Each of the item above is a link to the corresponding page in System Settings for the app, useful if the user wants to make any change. Links would work only in the PRO version.

**Comment by TheRealMystic** [comment]
While some of these permissions do show up as icons under the apps in the Apps tab, it won't reveal the full picture at a glance. Even if you add all these permissions as icons there, user will still have to scroll through it to see the status of all these permissions. Here too, 'At a Glance' won't work.

**Comment by TheRealMystic** [comment]
Here's how it would look:

[Permissions](https://user-images.githubusercontent.com/54448474/179671239-29040658-e8ff-4312-9f5e-6405765edb66.png)


Should be fine on mobile screens. 

You can replace the icons in the summary with the same ones that you have used under 'Full list of permissions'.

**Comment by TheRealMystic** [comment]
If you implement the colour change requested here:

https://github.com/d4rken-org/mystic-permissions/issues/9#issuecomment-1188619355

Then this is how the App Permissions will look like:

[Permissions](https://user-images.githubusercontent.com/54448474/179677759-7dfdcac1-dfe6-464b-9f5f-fa60f2a52842.png)

The same green and orange authorization indicators will also be used in Summary.

**Comment by vertigo220** [comment]
This summary could be further simplified to make it more compact and easier to tell at a glance what permissions are allowed by getting rid of the multiple columns and checkboxes and x's and just do one of the following:

- list only the allowed permissions
- List all desired permissions but color-code them (not sure if it would be better to have allowed as green and not allowed as red or vice-versa)
- List all desired permissions but show the ones not allowed as crossed out, which would eliminate the potential confusion with colors

Also, perhaps allow the user to customize which permissions to show in the summary. For example, I don't really care about DND and PiP
